### PR TITLE
Handle sorting on multivalue attributes.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review

### CODE REVIEW

File: searchlib/src/vespa/searchlib/attribute/attrvector.hpp [hunk 1]
```diff
@@ -89,6 +91,36 @@ NumericDirectAttrVector(const vespalib::string & baseFileName)
...
 }
 
+template <typename F, typename B>
+long
+NumericDirectAttrVector<F, B>::onSerializeForAscendingSort(DocId doc, void* serTo, long available, const search::common::BlobConverter* bc) const
+{
+    if (!F::IsMultiValue()) {
+        return search::NumericDirectAttribute<B>::onSerializeForAscendingSort(doc, serTo, available, bc);
+    }
+    search::attribute::NumericSortBlobWriter<BaseType, true> writer;
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/attribute/attrvector.hpp:101]: Consider adding a helper function so lines 101-106 can be shared between this function and _onSerializeForDescendingSort()_.

File: searchlib/src/vespa/searchlib/attribute/multinumericattribute.hpp [hunk 2]
```diff
@@ -183,4 +184,28 @@ MultiValueNumericAttribute<B, M>::onInitSave(vespalib::stringref fileName)
         (std::move(guard), this->createAttributeHeader(fileName), this->_mvMapping);
 }
 
+template <typename B, typename M>
+long
+MultiValueNumericAttribute<B, M>::onSerializeForAscendingSort(DocId doc, void* serTo, long available, const common::BlobConverter*) const
+{
+    attribute::NumericSortBlobWriter<T, true> writer;
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/attribute/multinumericattribute.hpp:191]: Consider adding a helper function so lines 191-196 can be shared between this function and _onSerializeForDescendingSort()_.

File: searchlib/src/vespa/searchlib/attribute/multinumericenumattribute.hpp [hunk 3]
```diff
@@ -145,5 +146,29 @@ MultiValueNumericEnumAttribute<B, M>::getSearch(QueryTermSimple::UP qTerm,
     return std::make_unique<attribute::MultiNumericEnumSearchContext<T, M>>(std::move(qTerm), *this, this->_mvMapping.make_read_view(doc_id_limit), this->_enumStore);
 }
 
+template <typename B, typename M>
+long
+MultiValueNumericEnumAttribute<B, M>::onSerializeForAscendingSort(DocId doc, void* serTo, long available, const common::BlobConverter*) const
+{
+    attribute::NumericSortBlobWriter<T, true> writer;
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/attribute/multinumericenumattribute.hpp:153]: Consider adding a helper function so lines 153-158 can be shared between this function and _onSerializeForDescendingSort()_.

**Review by geirst — DISMISSED** [review]
👍

File: searchlib/src/vespa/searchlib/attribute/attrvector.hpp (hunk 1 above)
Comment (toregge) [reply searchlib/src/vespa/searchlib/attribute/attrvector.hpp]: Done.

File: searchlib/src/vespa/searchlib/attribute/multinumericattribute.hpp (hunk 2 above)
Comment (toregge) [reply searchlib/src/vespa/searchlib/attribute/multinumericattribute.hpp]: Done.

File: searchlib/src/vespa/searchlib/attribute/multinumericenumattribute.hpp (hunk 3 above)
Comment (toregge) [reply searchlib/src/vespa/searchlib/attribute/multinumericenumattribute.hpp]: Done.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**